### PR TITLE
Remove the 1000 files limit when syncing from Desktop, Obsidian clients

### LINKED ISSUE #573: 1000-file limit?
I tried to add my knowledge base (composed of text files) to Khoj to try it out, but I get an ECONNRESET error when I try to save the folder and upload the files:

```
data: { detail: 'Too many files. Maximum number of files is 1000.' }
```

I have about 4,200 small files in my knowledge base, which doesn't seem like a particularly unreasonable number to me. 

I tried to install the self-hosted version assuming this was a limitation of the free plan, but this doesn't appear to be the case, as it doesn't work there either. I can't find reference to this error message in the code. What is this limit about? Is it something imposed by Axios?

### PR DESCRIPTION
**Pull request by debanjum**
### Major
- 92c47eae Push 1000 files at a time from the Desktop client for indexing
- fca7a5ff Push 1000 files at a time from the Obsidian client for indexing
- 4ded32cc Test 1000 file upload limit to index/update API endpoint

### Minor
- 79f72cf5 Show relevant error msg in desktop app, e.g when can't connect to server
- a8bb433a Pass indexed filenames in API response for client validation
- 066e39a7 Collect files to index in single dict to simplify index/update controller

Resolves #573

### CODE REVIEW

File: src/interface/desktop/main.js [hunk 1]
```diff
@@ -184,49 +184,51 @@ function pushDataToKhoj (regenerate = false) {
...
+    })
+    .catch(error => {
+        console.error(error);
+        state["completed"] = false;
+        if (error?.response?.status === 429 && (win = BrowserWindow.getAllWindows()[0])) {
+            state["error"] = `Looks like you're out of space to sync your files. <a href="https://app.khoj.dev/config">Upgrade your plan</a> to unlock more space.`;
+        } else if (error?.code === 'ECONNREFUSED') {
+            state["error"] = `Could not connect to Khoj server. Ensure you can connect to it at ${error.address}:${error.port}.`;
+        } else {
+            state["error"] = `Sync was unsuccessful at ${currentTime.toLocaleTimeString()}. Contact team@khoj.dev to report this issue.`;
```
Comment (sabaimran) [inline src/interface/desktop/main.js:222]: Nice error handling 👍🏾

File: src/interface/obsidian/src/utils.ts [hunk 2]
```diff
@@ -31,39 +31,54 @@ function fileExtensionToMimeType (extension: string): string {
 export async function updateContentIndex(vault: Vault, setting: KhojSetting, lastSyncedFiles: TFile[], regenerate: boolean = false): Promise<TFile[]> {
     // Get all markdown, pdf files in the vault
     console.log(`Khoj: Updating Khoj content index...`)
-    const files = vault.getFiles().filter(file => file.extension === 'md' || file.extension === 'pdf');
-    const binaryFileTypes = ['pdf', 'png', 'jpg', 'jpeg']
+    const files = vault.getFiles().filter(file => file.extension === 'md' || file.extension === 'markdown' || file.extension === 'pdf');
+    const binaryFileTypes = ['pdf']
```
Comment (sabaimran) [inline src/interface/obsidian/src/utils.ts:35]: Nice fix

File: src/interface/obsidian/src/utils.ts [hunk 3]
```diff
@@ -31,39 +31,54 @@ function fileExtensionToMimeType (extension: string): string {
...
     // Add all files to index as multipart form data
-    const formData = new FormData();
+    const fileData = [];
     for (const file of files) {
         countOfFilesToIndex++;
         const encoding = binaryFileTypes.includes(file.extension) ? "binary" : "utf8";
         const mimeType = fileExtensionToMimeType(file.extension) + (encoding === "utf8" ? "; charset=UTF-8" : "");
         const fileContent = encoding == 'binary' ? await vault.readBinary(file) : await vault.read(file);
-        formData.append('files', new Blob([fileContent], { type: mimeType }), file.path);
+        fileData.push({blob: new Blob([fileContent], { type: mimeType }), path: file.path});
```
Comment (sabaimran) [inline src/interface/obsidian/src/utils.ts:46]: Why the change in the data format?

**Review by sabaimran — APPROVED** [review]
Exciting getting this unblocked 🌄

File: src/interface/obsidian/src/utils.ts (hunk 3 above)
Comment (debanjum) [reply src/interface/obsidian/src/utils.ts:46]: There's no change in data format sent as request payload to API. `fileData` variable is just a temporary variable collecting file content and path as a key, value pairs object for batch processing later. Before the HTTP request is sent, the data collected in the `fileData` is used to form the payload with `'files'` as the key (like previously)

File: src/interface/obsidian/src/utils.ts (hunk 2 above)
Comment (debanjum) [reply src/interface/obsidian/src/utils.ts:35]: Thanks

File: src/interface/desktop/main.js (hunk 1 above)
Comment (debanjum) [reply src/interface/desktop/main.js:222]: Thanks! :)

**Pull request merged**